### PR TITLE
Print running pods before SchedulerPredicates tests

### PR DESCRIPTION
**Pull request by gmarek**
cc @yujuhong

### CODE REVIEW

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -169,6 +169,22 @@ var _ = Describe("SchedulerPredicates [Serial]", func() {
 
 		err = waitForPodsRunningReady(api.NamespaceSystem, systemPodsNo, podReadyBeforeTimeout)
 		Expect(err).NotTo(HaveOccurred())
+
+		for _, node := range nodeList.Items {
+			Logf("\nLogging pods the kubelet thinks is on node %v before test", node.Name)
+			podList, err := GetKubeletPods(c, node.Name)
```
Comment (yujuhong) [inline test/e2e/scheduler_predicates.go]: This function hits the /runningpods endpoint to get the pods that are running on the node (i.e., have at least one alive container shown in `docker ps`). kubelet admits pods by looking at the "desired list of pods", which is exposed at /pods. In your case, you will want to hit the /pods endpoint.

Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: Done.

File: test/e2e/scheduler_predicates.go [hunk 2]
```diff
@@ -169,6 +169,28 @@ var _ = Describe("SchedulerPredicates [Serial]", func() {
...
+				continue
+			}
+			podList := &api.PodList{}
+			err = result.Into(podList)
+			if err != nil {
+				Logf("Unable to cast result to pods for node %v", node.Name)
+				continue
+			}
+			for _, p := range podList.Items {
+				Logf("%v started at %v (%d container statuses recorded)", p.Name, p.Status.StartTime, len(p.Status.ContainerStatuses))
```
Comment (yujuhong) [inline test/e2e/scheduler_predicates.go]: You may want to print namespace too to make it clear what test left this pod, etc.

File: test/e2e/scheduler_predicates.go [hunk 3]
```diff
@@ -169,6 +169,28 @@ var _ = Describe("SchedulerPredicates [Serial]", func() {
 
 		err = waitForPodsRunningReady(api.NamespaceSystem, systemPodsNo, podReadyBeforeTimeout)
 		Expect(err).NotTo(HaveOccurred())
+
+		for _, node := range nodeList.Items {
+			Logf("\nLogging pods the kubelet thinks is on node %v before test", node.Name)
```
Comment (yujuhong) [inline test/e2e/scheduler_predicates.go]: suggestion: turn this into a function that can be reused. Given that this PR is for debugging flaky test and is high priority, I'll let you decide if you want to do that.

Comment (wojtek-t) [inline test/e2e/scheduler_predicates.go]: +1

File: test/e2e/kubelet_stats.go [hunk 4]
```diff
@@ -406,6 +406,27 @@ func GetKubeletPods(c *client.Client, node string) (*api.PodList, error) {
...
+		return
+	}
+	podList := &api.PodList{}
+	err = result.Into(podList)
+	if err != nil {
+		Logf("Unable to cast result to pods for node %v", nodeName)
+		return
+	}
+	for _, p := range podList.Items {
+		Logf("%v from %v started at %v (%d container statuses recorded)", p.Name, p.Namespace, p.Status.StartTime, len(p.Status.ContainerStatuses))
```
Comment (wojtek-t) [inline test/e2e/kubelet_stats.go]: Can you please add pod namespace too?

**Comment by wojtek-t** [comment]
One minor nit.

File: test/e2e/kubelet_stats.go (hunk 4 above)
Comment (wojtek-t) [inline test/e2e/kubelet_stats.go]: never mind

**Pull request merged**